### ISSUE TITLE
Suggest `use self::Thing::OtherThing` when using `use Thing::OtherThing`

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust#34191]
This frequently happens for code like this:
```rust
mod m {
    enum MyEnum {
        MyVariant
    }
    
    // Import all variants:
    use MyEnum::*;
}

fn main() {}
```

Since this *will* compile when putting `m`s contents into the main module, this can be quite confusing (for example when refactoring some code into it's own module).

Currently, this yields ``error: unresolved import `MyEnum::*`. Maybe a missing `extern crate MyEnum`?``. When this case is detected, the ``Maybe a missing `extern crate MyEnum`?`` probably shouldn't be printed.

**Comment by jseyfried** [comment]
cc me

**Comment by euclio** [comment]
I'd like to work on this.

I'm thinking: if we detect that adding "self" to the import resolves it, then a help message suggesting that is added. Otherwise, we add a help message that suggests `extern crate`. Does that make sense?

**Comment by jseyfried** [comment]
@euclio Yeah, that makes sense -- I think that's a good idea.

I would check whether prefixing `self` fixes the import in `finalize_import` (`resolve_imports.rs`).
If `resolve_module_path` fails, try `resolve_module_path_from_root` with a root of `self.current_module` (what `self` resolves to).
If that succeeds, prefixing `self` would fix it and you can return ``Failed(Some(span, "Try prepending `self::`"))``.
